packages/router-plugin/src/core/code-splitter/compilers.ts

```diff
@@ -138,27 +143,51 @@ export function compileCodeSplitReferenceRoute(
 
             if (
               !(
-                path.node.callee.name === 'createRoute' ||
-                path.node.callee.name === 'createFileRoute'
+                path.node.callee.name === 'createFileRoute' ||
+                path.node.callee.name === 'createRootRoute' ||
+                path.node.callee.name === 'createRootRouteWithContext'
               )
             ) {
               return
             }
 
+            createRouteFn = path.node.callee.name
+
             function babelHandleReference(routeOptions: t.Node | undefined) {
               const hasImportedOrDefinedIdentifier = (name: string) => {
                 return programPath.scope.hasBinding(name)
               }
 
               if (t.isObjectExpression(routeOptions)) {
+                if (opts.deleteNodes && opts.deleteNodes.size > 0) {
+                  routeOptions.properties = routeOptions.properties.filter(
+                    (prop) => {
+                      if (t.isObjectProperty(prop)) {
+                        if (t.isIdentifier(prop.key)) {
+                          if (opts.deleteNodes?.has(prop.key.name as any)) {
+                            return false
+                          }
+                        }
+                      }
+                      return true
+                    },
+                  )
+                }
+                if (createRouteFn !== 'createFileRoute') {
```

Comment:
I think its worth adding a comment here for future adventurers stating that the code-splitting we're exiting early here since the `rootRoute` is never code-split.

Perhaps the exit condition should just check for `createRootRoute` or `createRootRouteWithContext`.